src/lib/dialog/dialog.spec.ts

```diff
@@ -141,6 +141,20 @@ describe('MdDialog', () => {
     });
   }));
 
+  it('should close a dialog and get back a result before it is closed', async(() => {
+    let dialogRef = dialog.open(PizzaMsg, { viewContainerRef: testViewContainerRef });
+    let beforeCloseCallback = jasmine.createSpy('beforeClose callback');
+
+    dialogRef.beforeClose().subscribe(beforeCloseCallback);
+    dialogRef.close('Bulbasaurus');
+    viewContainerFixture.detectChanges();
+
+    viewContainerFixture.whenStable().then(() => {
+      expect(beforeCloseCallback).toHaveBeenCalledWith('Bulbasaurus');
+      expect(overlayContainerElement.querySelector('md-dialog-container')).toBeNull();
```

Comment:
This test checks that `beforeClose` occurred, but not _when_ it occurred. I'd restructure to be like
```ts
let beforeCloseInvoked = false;
const beforeCloseHandler = () => {
  expect(overlayContainerElement.querySelector('md-dialog-container')).toBeDefined();
  beforeCloseInvoked = true;
});
...
viewContainerFixture.whenStable().then(() => {
  expect(beforeCloseCallback).toHaveBeenCalledWith('Bulbasaurus');
  expect(overlayContainerElement.querySelector('md-dialog-container')).toBeNull();
});
```